sensors/Makefile

```diff
@@ -11,19 +11,25 @@ include $(static-lib.mk)
 # Sensors SPI interface
 NAME := libsensors-spi
 LOCAL_HEADERS := sensors-spi.h
-LOCAL_HEADERS_DIR := nothing  # do not reinstall libsensors.h
+LOCAL_HEADERS_DIR := nothing  # do not reinstall libsensors/*.h
 LOCAL_SRCS := libsensors-spi.c
 DEPS := libzynq7000-gpio-msg libspi-msg
 include $(static-lib.mk)
 
 # Sensors Server
-SENSORS_ALL := imu/lsm9dsxx imu/mpu6000 mag/lsm9dsxx mag/lis2mdl baro/lps25xx baro/ms5611 gps/pa6h gps/ubx
-SENSORS_SIM := imu/imu_sim.c mag/mag_sim.c baro/baro_sim.c gps/gps_sim.c gps/gps_sim.c
-SENSORS_LOCAL := $(SENSORS_SIM) $(addsuffix .c, $(filter $(SENSORS_ALL), $(DEVICE_SENSORS)))
+
+#Find all local sensor files
+SENSOR_DIRS := imu/ mag/ baro/ gps/
+SENSOR_SRCS := $(patsubst sensors/%, %, $(wildcard $(addsuffix *.c, $(addprefix sensors/, $(SENSOR_DIRS)))))
+
+SENSORS_SIM := $(filter %sim.c, $(SENSOR_SRCS))
+SENSORS_HW := $(filter-out %sim.c, $(SENSOR_SRCS))
+SENSORS_LOCAL := $(SENSORS_SIM) $(addsuffix .c, $(filter $(patsubst %.c, %, $(SENSORS_HW)), $(DEVICE_SENSORS)))
```

Comment:
after some thinking:
- we have 2 separate mechanisms to select the drivers:
  - one by *removing* some `.c` files from the default codebase 
  - one by *adding* extra static libs at a linking stage
- maybe we should use one mechanism
  - compile current drivers into static libs
  - provide default static libs list which might be overriden on the project basis
  - (that way you're removing the driver from the binary by just not-including it in the libraries list)

Maybe we should do it now - as we're already introducing a breaking change?

Reply:
+1